core/domain/collection_services.py

```diff
@@ -281,15 +281,20 @@ def get_learner_collection_dict_by_id(
     # dict includes meta information about the exploration (ID and title).
     for collection_node in collection_dict['nodes']:
         summary = exp_summaries_dict.get(collection_node['exploration_id'])
+        if not summary:
+            raise Exception(
```

Comment:
Could you throw a ``utils.ValidationError`` instead? You could probably just use the same string as suggested for the error thrown in ``_save_collection()`` below. Also, consider adding a sentence indicating why this might have happened, e.g.:

```python
Expected collection to only reference valid explorations, but found an exploration with ID: %s (was the exploration deleted?)
```